### ISSUE TITLE
Installer Permission Check

### ISSUE THREAD

**Issue by JimMackin** [salesagility/SuiteCRM#165]
The permission check in the installer checks every single file inside SuiteCRM regardless of whether is is a SuiteCRM file. This can cause issues if, say the web user doesn't have access to .git or if there is a large folder structure inside the SuiteCRM folder (like a joomla install).

We still want this to be future proof however so we'll need to think of a way to exclude files but include possible future SuiteCRM additions (possibly a white/blacklist?).

**Comment by pgorod** [comment]
The other permissions check, in the UpgradeWizard, already does this using an exclusion list.

Here is a code snippet from `modules/UpgradeWizard/systemCheck.php`:

```
// add directories here that should be skipped when doing file permissions checks (cache/upload is the nasty one)
$skipDirs = array(
	$sugar_config['upload_dir'],
	'.svn',
	'.git',
);
$files = uwFindAllFiles(getcwd(), array(), true, $skipDirs);
```
That function `uwFindAllFiles` is defined in `modules/UpgradeWizard/uw_utils.php`

So maybe this could be applied to the installer permissions check, but it needs a careful look into what the current code is doing.

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Installer Permission Check](https://trello.com/c/QDXiEsmD/160-installer-permission-check)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
